frame/staking/src/lib.rs

```diff
@@ -3075,6 +3108,80 @@ impl<T: Config> Module<T> {
 		}).collect::<Vec<(T::AccountId, Exposure<_, _>)>>()
 	}
 
+	/// Process the output of the election.
+	///
+	/// This ensures enough validators have been elected, converts all supports to exposures and
+	/// writes them to the associated storage.
+	///
+	/// Returns `Err(())` if less than [`MinimumValidatorCount`] validators have been elected, `Ok`
+	/// otherwise.
+	// TWO_PHASE_NOTE: remove the dead code.
+	#[allow(dead_code)]
+	pub fn process_election(
+		flat_supports: sp_npos_elections::Supports<T::AccountId>,
+		current_era: EraIndex,
+	) -> Result<Vec<T::AccountId>, ()> {
+		let exposures = Self::collect_exposures(flat_supports);
+		let elected_stashes = exposures.iter().cloned().map(|(x, _)| x).collect::<Vec<_>>();
+
+		if (elected_stashes.len() as u32) <= Self::minimum_validator_count() {
+			log!(
+				warn,
+				"chain does not have enough staking candidates to operate for era {:?}",
+				current_era,
+			);
+			return Err(());
+		}
+
+		// Populate Stakers and write slot stake.
+		let mut total_stake: BalanceOf<T> = Zero::zero();
+		exposures.into_iter().for_each(|(stash, exposure)| {
+			total_stake = total_stake.saturating_add(exposure.total);
+			<ErasStakers<T>>::insert(current_era, &stash, &exposure);
+
+			let mut exposure_clipped = exposure;
+			let clipped_max_len = T::MaxNominatorRewardedPerValidator::get() as usize;
+			if exposure_clipped.others.len() > clipped_max_len {
+				exposure_clipped.others.sort_by(|a, b| a.value.cmp(&b.value).reverse());
```

Comment:
what does `.reverse()` do here?

Reply:
we want to sort from greater to lesser, but it seems I didn't wrote a test for it (for `select_and_update_validators`) :-/

Reply:
I was a bit out of loop about this and mostly moved this code around without changing the logic. Fell free the push to the PR or open a new one on top of this, if something is missing.

Reply:
Couldnt you just reverse the compare order rather than calling reverse? or is that not how `cmp` works?

Reply:
yes you can

Reply:
fwiw for the sake of keeping things separated I won't change this in this PR.

Reply:
(fwiw: there is a test in staking contrary to what I said.)